lib/Service/ApplicationService.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/**
+ * @author Baptiste Fotia <baptiste.fotia@arawa.fr> for Arawa (https://arawa.fr)
+ *
+ * GroupFolders
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\GroupFolders\Service;
+
+use OCA\GroupFolders\AppInfo\Application;
+use OCP\App\IAppManager;
+
+class ApplicationService {
+
+    /** @var IAppManager */
+    private $appManager;
+
+    public function __construct(IAppManager $appManager)
+    {
+        $this->appManager = $appManager;
+    }
+
+    /**
+     * Check that all apps that depend on Groupfolders are installed
+     * @return boolean true if all apps are installed, false otherwise.
+     */
+    public function checkAppsInstalled() {
```

Comment:
Same should probably be done in the IinitialState of the settings

Reply:
Aah. You mean the user has to define the app that will use Groupfolders from a Multiselect and check it on the back-end ?

Reply:
I think I resolved this issue. Please, look at my comment here : https://github.com/nextcloud/groupfolders/pull/2072#discussion_r982380393